### PR TITLE
Add better v/freeze implementation for certain core functions, generic fns

### PR DESCRIPTION
**Pull request by sritchie**
This PR allows `g/add` and friends to properly freeze as `+`.

Clojure core functions that mirror our generics now freeze into their generic symbols, instead of freezing acting as identity as it did before.

### CODE REVIEW

File: src/sicmutils/generic.cljc [hunk 1]
```diff
@@ -48,32 +48,36 @@
...
                     (str "generic " f ".\n\n" (first options))
                     (str "generic " f ))
-        options (if (string? (first options))
-                  (next options)
-                  options)
-        kwd-klass (fork :clj clojure.lang.Keyword :cljs 'cljs.core/Keyword)]
+        options   (if (string? (first options))
+                    (next options)
+                    options)
+        [attr options] (if (map? (first options))
```
Comment (sritchie) [inline src/sicmutils/generic.cljc:54]: multimethods can't hold metadata, as far as I know... but it's our macro, so I can pretend and snatch a single key out of here!

File: src/sicmutils/generic.cljc [hunk 2]
```diff
@@ -48,32 +48,36 @@
...
     `(do
        (defmulti ~f ~docstring v/argument-kind ~@options)
        (defmethod ~f [~kwd-klass] [k#]
-         ({:arity ~arity :name '~f} k#)))))
+         ({:arity ~arity :name '~name} k#)))))
 
 ;; Numeric functions.
-(def-generic-function add 2)
-(def-generic-function negate 1)
+(def-generic-function add 2    {:name +})
```
Comment (sritchie) [inline src/sicmutils/generic.cljc:65]: this is how you override the symbol that gets returned by `v/freeze`.

File: test/sicmutils/function_test.cljc [hunk 3]
```diff
@@ -177,19 +185,19 @@
       (is (= 10 ((g/+ mul3 4) 2)))
       (is (= 32 ((g/expt 2 add2) 3)))
       (is (= 25 ((g/expt add2 2) 3)))
-      (is (= ::v/function (v/kind (g/expt add2 2))))
-
-      (testing "determinant"
```
Comment (sritchie) [inline test/sicmutils/function_test.cljc:182]: whoops, these were nested into another `testing` block.

File: test/sicmutils/function_test.cljc [hunk 4]
```diff
@@ -131,14 +147,6 @@
            (let [deferred (g/lcm (g/+ 1 g/square) 6)]
              (deferred 3))))))
 
-(defn transpose-defining-relation [T g a]
```
Comment (sritchie) [inline test/sicmutils/function_test.cljc:134]: this lives in `abstract.function-test` now.

**Review by littleredcomputer — APPROVED** [review]

**Pull request merged**